### PR TITLE
Feature/multiline code fences

### PR DESCRIPTION
**Pull request by bcanzanella**
`excludeParagraphWrap` does exclude the `<p>` tag, but it also prevents html highlighting (prettyPrint) from working

### CODE REVIEW

File: shared/ui/utils.ts [hunk 1]
```diff
@@ -384,6 +384,9 @@ export function escapeHtml(text: string) {
  */
 export function replaceHtml(text: string) {
 	const domParser = new DOMParser();
+	//input text's newlines will be created with <div> or <br> tags
+	//remove extra \n or \r\n to remove double lines later in markdown
+	text = text.replace(/\r\n/g, "").replace(/\n/g, "");
```
Comment (bcanzanella) [inline shared/ui/utils.ts:389]: this fixes the weird "dance" that previewing markdown would do, where additional empty lines would be inserted.

File: shared/ui/Stream/Codemark.tsx [hunk 2]
```diff
@@ -340,7 +340,7 @@ export class Codemark extends React.Component<Props, State> {
 		let index = 0;
 		this.skipMarkers = [];
 		while (index < groups.length) {
-			blocks.push(<MarkdownText text={groups[index]} excludeParagraphWrap={true} />);
+			blocks.push(<MarkdownText text={groups[index]} excludeParagraphWrap={false} />);
```
Comment (bcanzanella) [inline shared/ui/Stream/Codemark.tsx:343]: while `excludeParagraphWrap={true}` does remove the wrapping `<p>` tag, it also means use the `renderInline` function which prevents the html highlighting from working. 

To make this clearer, i was thinking of renaming that property to be named `inline`.

thoughts?

Comment (ppezaris) [reply shared/ui/Stream/Codemark.tsx:343]: sounds reasonable to me... might be worth adding a comment as documentation

Comment (ppezaris) [reply shared/ui/Stream/Codemark.tsx:343]: for this change, why not just remove excludeParagraphWrap=[whatever] since it defaults to false?

**Comment by ppezaris** [comment]
other than my comments LGTM

File: shared/ui/Stream/Codemark.tsx (hunk 2 above)
Comment (bcanzanella) [reply shared/ui/Stream/Codemark.tsx:343]: sounds good, I'll do that property rename and remove the `excludeParagraphWrap={false}` instances

**Review by ppezaris — APPROVED** [review]

**Pull request merged**